qcengine/programs/turbomole/runner.py

```diff
@@ -0,0 +1,214 @@
+            # We use basically a dummy stdin as we dont want to pipe any real
+            # input into define. We only want to parse the version number from
+            # the string.
+            with temporary_directory(suffix="_define_scratch") as tmpdir:
+                tmpdir = Path(tmpdir)
+                stdout = execute_define("\n", cwd=tmpdir)
+            # Tested with V7.3 and V7.4.0
+            version_re  = re.compile("TURBOMOLE (?:rev\. )?(V.+?)\s+")
+            mobj = version_re.search(stdout)
+            version = mobj[1]
+            self.version_cache[which_prog] = safe_version(version)
+        return self.version_cache[which_prog]
+
+    def compute(self, input_model: 'ResultInput', config: 'JobConfig') -> 'Result':
+        self.found(raise_error=True)
+
+        job_inputs = self.build_input(input_model, config)
+        success, dexe = self.execute(job_inputs)
+
+        # TODO: handle input errors?! But then define probably already crashed...
+        # if 'There is an error in the input file' in dexe["stdout"]:
+            # raise InputError(dexe["stdout"])
+
+        if success:
+            dexe["outfiles"]["stdout"] = dexe["stdout"]
+            dexe["outfiles"]["stderr"] = dexe["stderr"]
+            return self.parse_output(dexe["outfiles"], input_model)
+
+    def build_input(self, input_model: 'ResultInput', config: 'JobConfig',
+                    template: Optional[str] = None) -> Dict[str, Any]:
+        turbomolrec = {
+            'infiles': {},
+            'outfiles': {"control": "control", },
+            'scratch_directory': config.scratch_directory,
+        }
+
+        # Handle molecule
+        # TODO: what's up with moldata? Do I need it?
+        coord_str, moldata = input_model.molecule.to_string(dtype='turbomole', return_data=True)
+
+        # Prepare stdin for define call
+        model = input_model.model
+        # geeopt will hold the for which to calculate the gradient.
+        # 'x' corresponds to the ground state, 'a 1' would be the GS too.
+        # 'a1 2' would be the 1st excited state of the irreducible group A1.
+        # Right now only GS are supported, so this is hardcoded as 'x'.
+        geoopt = "x" if input_model.driver.derivative_int() > 0 else ""
+        stdin = prepare_stdin(model.method, model.basis, input_model.keywords,
+                              input_model.molecule.molecular_charge,
+                              input_model.molecule.molecular_multiplicity,
+                              geoopt,
+        )
+        with temporary_directory(suffix="_define_scratch") as tmpdir:
+            tmpdir = Path(tmpdir)
+            with open(tmpdir / "coord", "w") as handle:
+                handle.write(coord_str)
+            stdout = execute_define(stdin, cwd=tmpdir)
+            # The define scratch will be populated by some files that we want to keep
+            to_keep = "basis auxbasis coord control alpha beta mos".split()
```

Comment:
Can we switch the splits to explicit lists when you have the chance? Or is this easy to do from the docs?

Reply:
fixed in d0ddc48